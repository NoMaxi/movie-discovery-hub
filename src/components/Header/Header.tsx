import bgHeaderImage from "@/assets/bg-header.png";
import { Outlet, useParams } from "react-router-dom";

export const Header = () => {
    const { movieId } = useParams();
    const heightClass = movieId ? "h-[540px]" : "min-h-[290px]";

    return (
        <header
            data-testid="header"
            className="
                relative w-[var(--content-width)] px-[60px] pt-[80px] pb-[20px] mb-[10px]
                bg-[var(--color-content-background)] bg-cover bg-center bg-no-repeat
            "
            style={{ backgroundImage: `url(${bgHeaderImage})` }}
        >
            <div data-testid="header-inner" className={`flex flex-col relative z-10 ${heightClass}`}>
                <Outlet data-testid="outlet-inner" />
            </div>
        </header>
    );
};
