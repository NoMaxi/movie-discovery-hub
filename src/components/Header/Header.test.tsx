import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import { Header } from "./Header";

jest.mock("react-router-dom", () => {
    const actual = jest.requireActual("react-router-dom");
    return {
        ...actual,
        Outlet: () => <div data-testid="outlet" />,
        useParams: jest.fn(),
    };
});

const { useParams } = jest.requireMock("react-router-dom");

describe("Header", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    const renderWithRouter = (movieId?: string) => {
        (useParams as jest.Mock).mockReturnValue(movieId ? { movieId } : {});
        return render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>,
        );
    };

    test("Should render a header with the correct background and base classes", () => {
        const { asFragment } = renderWithRouter();
        const header = screen.getByTestId("header");

        expect(header).toBeInTheDocument();
        expect(header).toHaveStyle({ backgroundImage: expect.stringContaining("bg_header.png") });
        expect(header).toHaveClass("relative");
        expect(header).toHaveClass("w-[var(--content-width)]");
        expect(asFragment()).toMatchSnapshot();
    });

    test("Should apply 'min-h-[290px]' class when 'movieId' is not present in URL params", () => {
        renderWithRouter();
        expect(screen.getByTestId("header-inner")).toHaveClass("min-h-[290px]");
    });

    test("Should apply 'h-[540px]' class when 'movieId' is present in URL params", () => {
        renderWithRouter("123");
        expect(screen.getByTestId("header-inner")).toHaveClass("h-[540px]");
    });

    test("Should render Outlet", () => {
        renderWithRouter();
        expect(screen.getByTestId("outlet")).toBeInTheDocument();
    });
});
