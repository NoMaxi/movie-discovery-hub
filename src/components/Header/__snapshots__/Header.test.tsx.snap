// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render a header with the correct background and base classes 1`] = `
<DocumentFragment>
  <header
    class="
                relative w-[var(--content-width)] px-[60px] pt-[80px] pb-[20px] mb-[10px]
                bg-[var(--color-content-background)] bg-cover bg-center bg-no-repeat
            "
    data-testid="header"
    style="background-image: url();"
  >
    <div
      class="flex flex-col relative z-10 min-h-[290px]"
      data-testid="header-inner"
    >
      <div
        data-testid="outlet"
      />
    </div>
  </header>
</DocumentFragment>
`;
